### PR TITLE
Remove BENCHMARK_ENABLE_SHARED and prefer BUILD_SHARED_LIBS

### PR DESCRIPTION
**Pull request by EricWF**
Address comments from #114.

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -17,13 +17,12 @@ else()
   message(FATAL_ERROR "Failed to determine the source files for the regular expression backend")
 endif()
 
+add_library(benchmark ${SOURCE_FILES} ${RE_FILES})
+
 # Build the benchmark library
```
Comment (mattyclarkson) [inline src/CMakeLists.txt]: This comment would make more sense on line 19. A more apt comment would be `# Link threading if building shared library`

Comment (EricWF) [inline src/CMakeLists.txt]: ack.

**Comment by mattyclarkson** [comment]
@EricWF, FYI if you add `resolves #114` in either the PR description or the commit message it will close the associated issue when you merge the PR.

**Comment by dominichamon** [comment]
LGTM

**Pull request merged**